### PR TITLE
Add body make options for client & server so clients can define how to handle values

### PR DESCRIPTION
**Pull request by smartly-petri**
Clients of `client` & `server` implementations can define what to do with unknown fields; to throw error or just drop fields.

### CODE REVIEW

File: packages/oats/src/generate-server.ts [hunk 1]
```diff
@@ -498,6 +498,42 @@ export function generateCreateRouter() {
...
+                            'parseNumericStrings',
+                            'bodyParseNumericStrings',
+                            ts.factory.createTrue()
+                          ),
+                          ts.factory.createBindingElement(
+                            undefined,
+                            'allowConvertForArrayType',
+                            'bodyAllowConvertForArrayType',
+                            ts.factory.createTrue()
+                          ),
```
Comment (buttilda) [inline packages/oats/src/generate-server.ts:528]: i don't think we need to deconstruct these

Comment (smartly-petri) [reply packages/oats/src/generate-server.ts]: Yeah, no need at least for now. Removed 👍

File: packages/oats/src/generate-server.ts [hunk 2]
```diff
@@ -565,7 +601,29 @@ export function generateCreateRouter() {
...
+                          ts.factory.createIdentifier('bodyParseBooleanStrings')
+                        ),
+                        ts.factory.createPropertyAssignment(
+                          'parseNumericStrings',
+                          ts.factory.createIdentifier('bodyParseNumericStrings')
+                        ),
+                        ts.factory.createPropertyAssignment(
+                          'allowConvertForArrayType',
+                          ts.factory.createIdentifier('bodyAllowConvertForArrayType')
+                        )
```
[the remark this answers is not in the document]
Comment (smartly-petri) [reply packages/oats/src/generate-server.ts]: Done

**Review by buttilda — APPROVED** [review]
LGTM

**Pull request merged**